server/service/devices.go

```diff
@@ -722,6 +806,18 @@ func getDeviceSoftwareEndpoint(ctx context.Context, request interface{}, svc fle
 
 	req := request.(*getDeviceSoftwareRequest)
 	res, meta, err := svc.ListHostSoftware(ctx, host.ID, req.HostSoftwareTitleListOptions)
+	for _, s := range res {
+		// convert api style iconURL to device token URL
+		if s.IconUrl != nil && *s.IconUrl != "" {
+			matched := softwareTitleIconURLRegex.MatchString(*s.IconUrl)
+			if matched {
+				icon := fleet.SoftwareTitleIcon{SoftwareTitleID: s.ID}
+				deviceIconURL := icon.IconUrlWithDeviceToken(req.Token)
+				s.IconUrl = ptr.String(deviceIconURL)
+			}
+		}
+	}
+
```

Comment:
really don't like this here, but we need the `req.token`, and I couldn't figure out how to get this in the service without adding an extra param.

Reply:
So the code here would turn into?
```
for _, s := range res {
  s.ForMyDevicePage(req.Token)
}
```

I guess at least it would be easy to write a test in isolation for the logic if it was on the `HostSoftwareWithInstaller` struct.

Reply:
Yeah, that's what I was thinking.